### DISCUSSION TITLE
Systemctl does not register dnscrypt-proxy.service on Ubuntu

### DISCUSSION THREAD

**Discussion by Beeblebrox-BSD** [DNSCrypt/dnscrypt-proxy#2510] [Q&A]
UNAME: Ubuntu Luna, 6.2.0-34-generic https://github.com/DNSCrypt/dnscrypt-proxy/issues/34-Ubuntu SMP PREEMPT_DYNAMIC x86_64 x86_64 x86_64 GNU/Linux

./dnscrypt-proxy -version: 2.0.45

./dnscrypt-proxy -check:
dnscrypt-proxy 2.0.45
Source [public-resolvers] loaded
Configuration successfully checked

./dnscrypt-proxy -config ./dnscrypt-proxy.toml
dnscrypt-proxy 2.0.45
Network connectivity detected
Source [public-resolvers] loaded
Firefox workaround initialized
[cloudflare] OK (DoH) - rtt: 57ms
Server with the lowest initial latency: cloudflare (rtt: 57ms)
dnscrypt-proxy is ready - live servers: 1

FROM CLEAN INSTALL OUTPUT:
Created symlink /etc/systemd/system/multi-user.target.wants/dnscrypt-proxy-resolvconf.
service → /lib/systemd/system/dnscrypt-proxy-resolvconf.service.
Created symlink /etc/systemd/system/sockets.target.wants/dnscrypt-proxy.socket → /lib/
systemd/system/dnscrypt-proxy.socket.
Created symlink /etc/systemd/system/multi-user.target.wants/dnscrypt-proxy.service → /
lib/systemd/system/dnscrypt-proxy.service.
Could not execute systemctl: at /usr/bin/deb-systemd-invoke line 145.

SYSTEMCTL DEBUG ATTEMPT:
1.    sudo systemctl start dnscypt-proxy
    Failed to start dnscypt-proxy.service: Unit dnscypt-proxy.service not found.

2.    cd /lib/systemd/system/ && sudo systemd-analyze verify ./dnscypt-proxy.service
    Unit dnscypt-proxy.service not found.

3.    cd /etc/systemd/system/multi-user.target.wants && sudo systemd-analyze verify ./dnscypt-proxy.service
    Unit dnscypt-proxy.service not found.

4.    passwd file has: `_dnscrypt-proxy:x:131:65534::/run/dnscrypt-proxy:/usr/sbin/nologin`

**Comment by MatzFan** [comment]
Had the same installation issue (`Could not execute systemctl:  at /usr/bin/deb-systemd-invoke line 145.`) on LMDE 6 after installing `dnscrypt-proxy` from Debian testing (it's not available for bookworm). A reboot resolved the issue:
```
$ systemctl --type=service --state=running|grep dnscrypt-proxy
  dnscrypt-proxy.service     loaded active running DNSCrypt client proxy
```

**Reply by MatzFan** [reply]
After a bit of debugging it appears line 145 in `/usr/bin/deb-systemd-invoke` is trying to execute this command: `systemctl --quiet --system start dnscrypt-proxy-resolvconf.service dnscrypt-proxy.socket`. If I try to run that command myself less the `--quiet` switch I get
```bash
$ sudo systemctl --system start dnscrypt-proxy-resolvconf.service dnscrypt-proxy.socket
Job for dnscrypt-proxy-resolvconf.service failed because the control process exited with error code.
See "systemctl status dnscrypt-proxy-resolvconf.service" and "journalctl -xeu dnscrypt-proxy-resolvconf.service" for details.
```
If I do that I see this:
```bash
$ systemctl status dnscrypt-proxy-resolvconf.service
× dnscrypt-proxy-resolvconf.service - DNSCrypt proxy resolvconf support
     Loaded: loaded (/lib/systemd/system/dnscrypt-proxy-resolvconf.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Thu 2024-09-12 13:52:11 BST; 40s ago
       Docs: https://github.com/DNSCrypt/dnscrypt-proxy/wiki
    Process: 9047 ExecStart=/bin/sh -c systemctl show dnscrypt-proxy.socket                      | grep "Listen.*Datagram"                      | c>
   Main PID: 9047 (code=exited, status=1/FAILURE)
        CPU: 21ms

Sep 12 13:52:11 2 systemd[1]: Starting dnscrypt-proxy-resolvconf.service - DNSCrypt proxy resolvconf support...
Sep 12 13:52:11 2 resolvconf[9053]: Dropped protocol specifier '.dnscrypt-proxy' from 'lo.dnscrypt-proxy'. Using 'lo' (ifindex=1).
Sep 12 13:52:11 2 resolvconf[9053]: Failed to set DNS configuration: Unit dbus-org.freedesktop.network1.service not found.
Sep 12 13:52:11 2 systemd[1]: dnscrypt-proxy-resolvconf.service: Main process exited, code=exited, status=1/FAILURE
Sep 12 13:52:11 2 systemd[1]: dnscrypt-proxy-resolvconf.service: Failed with result 'exit-code'.
Sep 12 13:52:11 2 systemd[1]: Failed to start dnscrypt-proxy-resolvconf.service - DNSCrypt proxy resolvconf support.
```
`Unit dbus-org.freedesktop.network1.service not found.` seems to be the cause of the failure. Not sure why this fails, but I hope this helps fix the original issue.
